### PR TITLE
Clarify how to make predictive in SVI

### LINKED ISSUE #1309: Correct SVI API usage
So I was playing around with SVI API and problems from Statistical Rethinking 2ed. And one thing - extracting posterior sample data - made me quite puzzled for a while.

The [doc on SVI](https://num.pyro.ai/en/stable/svi.html#numpyro.infer.svi.SVI) describes intended usage as:
```python
# ...
# get posterior samples
predictive = Predictive(guide, params=params, num_samples=1000)
samples = predictive(random.PRNGKey(1), data)
```

The result of that for my model was: 
1. Sample with deterministic estimands missing if I used `Predictive(guide, params=params, num_samples=1000)`
```python
>>> samples.keys()
# actual: dict_keys(['alpha', 'beta', 'sigma'])
# expected: dict_keys(['W', 'alpha', 'beta', 'mu', 'sigma'])
```
2. Empty sample if I used `Predictive(guide, params=params, num_samples=1000)`
```python
>>> samples.keys()
# actual: dict_keys([])
# expected: dict_keys(['W', 'mu'])
```
---
Changing a snippet like this solved the issue:
```python
# ...
# get posterior samples
predictive = Predictive(guide.model, params=params, num_samples=1000)
samples = predictive(random.PRNGKey(1), data)
```
I've found this solution while comparing documentation code to "Code 4.53" by @fehiepsi from [here](https://github.com/fehiepsi/rethinking-numpyro/blob/master/notebooks/04_geocentric_models.ipynb ) 

---

My full model code:
```python
import warnings

warnings.filterwarnings("ignore", category=DeprecationWarning)

import pandas as pd

import jax.numpy as jnp
from jax import random

import numpyro
import numpyro.distributions as dist
import numpyro.optim as optim
from numpyro.infer import SVI, Trace_ELBO, Predictive
from numpyro.infer.autoguide import AutoNormal

# INFO: load data for problems from the official repo
data_uri = "https://raw.githubusercontent.com/rmcelreath/rethinking/master/data/Howell1.csv"
df_dev = pd.read_csv(data_uri, sep=";")
df_dev = df_dev[df_dev["age"] >= 18]

def model(df_dev):
    alpha = numpyro.sample("alpha", dist.Normal(60, 10))
    beta = numpyro.sample("beta", dist.LogNormal(0, 1))
    sigma = numpyro.sample("sigma", dist.Uniform(0, 10))
    mu = numpyro.deterministic("mu", alpha + beta * (df_dev["height"] - df_dev["height"].mean()).values)
    numpyro.sample("W", dist.Normal(mu, sigma), obs=df_dev["weight"].values)



# quadratic approximation part
guide = AutoNormal(model)
svi = SVI(model, guide, optim.Adam(1), Trace_ELBO(), df_dev=df_dev)
svi_result = svi.run(random.PRNGKey(0), 1000)

params = svi_result.params

# display summary of quadratic approximation
samples = guide.sample_posterior(random.PRNGKey(1), params, (1000,))
numpyro.diagnostics.print_summary(samples, prob=0.89, group_by_chain=False)

# INFO: bad
predictive = Predictive(guide, params=params, posterior_samples=samples)
samples_predictive = predictive(random.PRNGKey(1), df_dev)
print(samples_predictive.keys())

# INFO: good
predictive = Predictive(guide.model, params=params, posterior_samples=samples)
samples_predictive = predictive(random.PRNGKey(1), df_dev)
print(samples_predictive.keys())

```
---
Please tell me what do you think

### PR DESCRIPTION
**Pull request by fehiepsi**
Fixes #1309

Based on @ColdTeapot273K work at #1363

The original PR #1363 has been pending for response for a while. So it might be good to have this PR closing the issue.

### CODE REVIEW

File: numpyro/infer/svi.py [hunk 1]
```diff
@@ -110,9 +110,15 @@ class SVI(object):
...
         >>> inferred_mean = params["alpha_q"] / (params["alpha_q"] + params["beta_q"])
+        >>> # use guide to make predictive
+        >>> predictive = Predictive(model, guide=guide, params=params, num_samples=1000)
+        >>> samples = predictive(random.PRNGKey(1), data=None)
         >>> # get posterior samples
         >>> predictive = Predictive(guide, params=params, num_samples=1000)
-        >>> samples = predictive(random.PRNGKey(1), data)
+        >>> posterior_samples = predictive(random.PRNGKey(1), data=None)
+        >>> # use posterior samples to make predictive
+        >>> predictive = Predictive(model, guide=guide, params=params, num_samples=1000)
```
Comment (ordabayevy) [inline numpyro/infer/svi.py:120]: What is the difference between lines 114 and 120? They look identical.

Comment (ordabayevy) [reply numpyro/infer/svi.py:120]: Was it supposed to be:
```py
predictive = Predictive(model, posterior_samples=posterior_samples, params=params, num_samples=1000)
```

Comment (fehiepsi) [reply numpyro/infer/svi.py]: thanks, you're right

**Review by ordabayevy — APPROVED** [review]
LGTM now!

**Pull request merged**